pkg/tfpfbridge/tests/internal/testprovider/testbridge_resource_testres.go

```diff
@@ -503,19 +506,36 @@ func (e *testres) Delete(ctx context.Context, req resource.DeleteRequest, resp *
 }
 
 func (e *testres) freshID(statedir string) (string, error) {
-	i := 0
-	for {
-		candidate := fmt.Sprintf("%d", i)
-		_, err := os.Stat(e.cloudStateFile(statedir, candidate))
-		if os.IsNotExist(err) {
-			return candidate, nil
+	mu := fsutil.NewFileMutex(filepath.Join(statedir, "testres.lock"))
+	if err := mu.Lock(); err != nil {
+		return "", err
+	}
+	defer func() {
+		if err := mu.Unlock(); err != nil {
+			panic(err)
 		}
+	}()
+
+	cF := filepath.Join(statedir, "testres.counter")
```

Comment:
is `cF` the location of the resource file?

Reply:
It's a special counterFile that stores the integer that we use as a resource ID. We increment it and store to disk. This has to work across processes. A mini database of sorts.